### DISCUSSION TITLE
Authorization stops working after 2 weeks

### DISCUSSION THREAD

**Discussion by nordinho** [amzn/selling-partner-api-models#3036] [General]
I'm using a python module called AMAZON-SP-API to connect to the API. I successfully authorized myself to connect to the API and managed to pull data from it a few weeks ago, but now when I try to connect again with the same code I receive this message:

    raise exception(error, headers=res.headers)
sp_api.base.exceptions.SellingApiForbiddenException: [{'message': 'Access to requested resource is denied.', 'code': 'Unauthorized', 'details': ''}]

I have no idea why this is happening, I tried to get a new refresh token but it's not working. This happened before and I just created a new app and  IAM role and it worked, but now 2 weeks later and the authorization isn't working again.

Any ideas?

**Comment by dane5102** [comment]
Have you changed the IAM ARN for the app recently? When you change it, the original IAM ARN you used is still active/usable for 2 weeks. After that, the original becomes inactive and uses only the new one you updated it to. It's possible your code was working under the previous IAM ARN you had, but after the 2 weeks it became inactive and started using the new IAM ARN you input.

I had a similar problem where I changed the IAM ARN from User to Role because my code wasn't working. After I changed it, I modified my code a bit and it started working for exactly two weeks and then I started getting unauthorized errors. When I tried switching around the IAM ARN, it mentions how the old IAM ARN is still active for 2 weeks before not being usable. It seemed to be that whatever I modified after changing it to Role was actually working for User and since it was still active for 2 weeks, it looked like my code was working for Role but was actually working under User, hence why it stopped working for me after 2 weeks.

**Reply by leegis123** [reply]
That's exactly what happened to me. I changed it back from Role to User now but the error is still there. Did changing it back resolve it for you or did you have to do something else?
